chia/farmer/farmer_api.py

```diff
@@ -305,6 +309,10 @@ async def respond_signatures(self, response: harvester_protocol.RespondSignature
         if response.sp_hash not in self.farmer.sps:
             self.farmer.log.warning(f"Do not have challenge hash {response.challenge_hash}")
             return None
+        filter_prefix_bits = self.farmer.sp_hash_to_filter_size.get(response.sp_hash)
```

Comment:
> If we stick with this new cache and it seems there may be more than one signage point in the `sps` for a given `sp_hash`. Are you sure `sp_hash_to_filter_size.get(response.sp_hash)` contains what we really want at this point and can't be overwritten when another signage point gets added to the list `sps[sp_hash]`?

Hm or actually even then it should be for the same height/prefix_bits and we can just pick one of them, no?

Reply:
I was sure that we were not guaranteed to have access to signage point message at the time we needed to check a proof. But I can't find it, so maybe I was mistaken. @AmineKhaldi would you mind investigating and make this change it it works?

Reply:
Pretty sure this should work since the if block above checks that `response.sp_hash` is in `Farmer.sps` and here https://github.com/Chia-Network/chia-blockchain/blob/ab0c22bed72283ba8e5566e72c3d6027f39a6ed9/chia/farmer/farmer_api.py#L318 we already access the first entry of it.